source/common/stats/utility.h

```diff
@@ -30,6 +30,25 @@ class DynamicName : public absl::string_view {
   explicit DynamicName(absl::string_view str) : absl::string_view(str) {}
 };
 
+/**
+ * Represents a dynamically created stat name token based on std::string.
+ * This class wrapper is used in the 'Element' variant so that call-sites
+ * can express explicit intent to create dynamic stat names, which are more
+ * expensive than symbolic stat names. We use dynamic stat names only for
+ * building stats based on names discovered in the line of a request.
+ *
+ * Specifically, this class should be used only when the content of the string
+ * can be changed between the object creation and usage.
+ */
+class DynamicSavedName : public std::string {
+public:
+  // This is intentionally left as an implicit conversion from string_view to
+  // make call-sites easier to read, e.g.
+  //    Utility::counterFromElements(*scope, {DynamicName("a"), DynamicName("b")});
+  explicit DynamicSavedName(const std::string& str) : std::string(str) {}
```

Comment:
Can we drop this constructor?

Reply:
it seems like we can, it works even better, thanks